### DISCUSSION TITLE
Configuration structure with stand-alone executable

### DISCUSSION THREAD

**Discussion by takase1121** [keplerproject/luarocks#1582] [Q&A] [answered]
I tried to install luarocks stand-alone executable to my lua 5.2 installation.
However, I cannot figure out why luarocks cannot find my config.
Apparently, the standalone installation always defaults to 5.3.

My question is, why luarocks does not detect other config files (`config-*.lua`) and provide a list and (optionally) allow user to set version when they install a rock?

**Comment by hishamhm** [comment, marked as answer]
if you type `luarocks config lua_version 5.2` it should switch to Lua 5.2.

Also, if you run `luarocks` with no arguments it should list you the locations where it is looking for configuration files. You can also set the `LUAROCKS_CONFIG` environment variable to override it.

Hope this helps!
